harbor-client/src/lightning_address.rs

```diff
@@ -0,0 +1,45 @@
+use crate::http::{make_get_request_direct, make_get_request_tor};
+use lnurl::{
+    lightning_address::LightningAddress,
+    lnurl::LnUrl,
+    pay::{LnURLPayInvoice, PayResponse},
+};
+use std::str::FromStr;
+
+pub fn parse_lightning_address(address: &str) -> anyhow::Result<LightningAddress> {
+    let ln_address = LightningAddress::from_str(address)?;
+    Ok(ln_address)
+}
+
+pub async fn get_invoice(
+    pay: &PayResponse,
+    msats: u64,
+    tor_enabled: bool,
+) -> anyhow::Result<LnURLPayInvoice> {
+    if msats < pay.min_sendable || msats > pay.max_sendable {
+        return Err(anyhow::anyhow!("Invalid amount"));
+    }
+
+    let symbol = if pay.callback.contains('?') { "&" } else { "?" };
+    let url = format!("{}{}amount={}", pay.callback, symbol, msats);
+
+    if tor_enabled {
+        make_get_request_tor(&url).await
+    } else {
+        make_get_request_direct(&url).await
+    }
+}
+
+pub async fn make_lnurl_request(lnurl: &LnUrl, tor_enabled: bool) -> anyhow::Result<PayResponse> {
+    let lightning_address = lnurl
+        .lightning_address()
+        .ok_or(anyhow::anyhow!("No lightning address found"))?;
+    let lnurlp = lightning_address.lnurlp_url();
```

Comment:
should just get the url from the lnurl, converting to lightning address does nothing here but limit the amount of valid lnurls

Reply:
I couldn't figure that out unless I use the url lib to get it?

Reply:
nm I found it